### ISSUE TITLE
Crash on search

### ISSUE THREAD

**Issue by rbleuse** [mvysny/aedict#962]
Hi Martin,

I got a crash every time I type でんえん (french dictionary).

It appears when I type the last `n` from `dennen`

<img src="https://user-images.githubusercontent.com/626863/124714469-c4ef9700-df3c-11eb-939f-bcf4b0ac040a.jpg" width="300" /><img src="https://user-images.githubusercontent.com/626863/124714472-c6b95a80-df3c-11eb-9950-a7bc49ac77cf.jpg" width="300" />

**Comment by mvysny** [comment]
Uh-oh, there are two issues here:
* The "Dictionary File Problem" dialog should show the exception in more detail - I'll revisit the code and fix this.
* The deinflection engine is not filling the word-class info for "malformed" verbs. Aedict doesn't expect this and will fail. I need to think on whether the word-class (godan/ichidan/suru etc) can actually be empty. I don't think it makes much sense since if a deinflection rule was applied, then that deinflection rule should fill in the word-class. This feels to me like a bug in jputils deinflection engine. Let us take a look.

**Comment by mvysny** [comment]
@rbleuse just to be extra-sure: it's illegal to even attempt to deinflect でんえ to e.g. denu or something, right? :)

**Comment by rbleuse** [comment]
I didn't analyze much, but for example the only verb that ends with nu is `shinu`, which gives `shine` for the imperative form, so I think that the deinflection engine is right.
I don't know if the process is same for `でんえ` and `でね` though
Indeed, only `でね` should be deinflected into `でぬ`, not `でんえ`.

The second problem is that the engine translates `えんえ` into `exnu`, and that extra x makes that the deinflection is not found :

```
val result = EnumSet.noneOf(InflectableWordClass::class.java)
for (deinflection in verb[deinflected]) {
    for (def in deinflection.deinflected.filter { it.key == deinflected }.values) {
        result.addAll(def)
    }
}
```

The filter can't find `exnu` whereas the deinflected tree contains `enu`

**Comment by mvysny** [comment]
Yup, the `x` thing was caused the deinflection not to be found. I've fixed this in https://gitlab.com/mvysny/jputils/-/commit/54820e9759c69fc56f230df4a4978e2c587a0aa3 ; I'll close this ticket once new jputils has been released & integrated into aedict.

**Comment by mvysny** [comment]
Fixed in Aedict 3.50.22
